src/Request.php

```diff
@@ -132,7 +132,11 @@ public function handleData($data)
         $this->buffer .= $data;
 
         if (false !== strpos($this->buffer, "\r\n\r\n")) {
-            list($response, $bodyChunk) = $this->parseResponse($this->buffer);
+            try {
+                list($response, $bodyChunk) = $this->parseResponse($this->buffer);
+            } catch (\InvalidArgumentException $exception) {
+                $this->emit('error', [$exception, $this]);
+            }
```

Comment:
Sure, I just wasn't sure if we need to clear the buffer and remove listeners if we emit an error.
Similar changes in react/http made me think that they must be removed/buffer must be cleared even if the response parsing was not successful.
Isn't that the case here?